app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/group/GroupMemberGrsController.scala

```diff
@@ -142,16 +148,26 @@ class GroupMemberGrsController @Inject() (
     update { registration =>
       val updatedGroupDetails: GroupDetail = registration.groupDetail match {
         case Some(groupDetail) =>
-          val member = addGroupMember(details, orgType)
-          if (!groupDetail.memberAlreadyPresent(member)) {
+          val member: GroupMember = addGroupMember(details, orgType)
+          if (isMemberPresent(member, groupDetail, registration))
+            groupDetail
+          else {
             val members: Seq[GroupMember] = groupDetail.members :+ member
             groupDetail.copy(members = members, currentMemberOrganisationType = None)
-          } else
-            groupDetail
-
-        case None => throw new InternalServerException(s"No group detail")
+          }
+        case None => throw new IllegalStateException(s"No group detail")
       }
       registration.copy(groupDetail = Some(updatedGroupDetails))
     }
 
+  def isMemberPresent(
+    member: GroupMember,
+    groupDetail: GroupDetail,
+    registration: Registration
+  ): Boolean =
+    groupDetail.members.exists(groupMember => groupMember.equals(member)) ||
```

Comment:
Can you just use `.contains()` when equals is available?

Reply:
No we cant use contains method. As it is a member and not a list you cant use contains.